src/shim/bo.cpp

```diff
@@ -8,7 +8,7 @@
 namespace {
 
 uint32_t
-alloc_drm_bo(const shim_xdna::pdev& dev, amdxdna_bo_type type, void* buf, size_t size)
+alloc_drm_bo(const shim_xdna::pdev& dev, uint32_t type, void* buf, size_t size)
```

Comment:
I'd keep the amdxdna_bo_type here for this api even it is prviate anonymous namespace.

And cast it to u32 at line16. Does that make sense?

Reply:
> I tried it, the compiler doesn't like it it won't build that way.

this should work:
.type = static_cast<uint32_t>(type),